config/environments/production.rb

```diff
@@ -30,3 +30,45 @@
 gsub_file "config/environments/production.rb",
           "config.log_level = :debug",
           'config.log_level = ENV.fetch("LOG_LEVEL", "info").to_sym'
+
+insert_into_file "config/environments/production.rb",
+  after: /.*config\.public_file_server\.enabled.*\n/ do
+  <<~'RUBY'
+
+    # Ensure that Rails sets appropriate caching headers on static assets if
+    # Rails is serving static assets in production e.g. on Heroku
+    #
+    # | Seconds  | Days     |
+    # |----------|----------|
+    # | 86400    | 1 day    |
+    # | 15552000 | 180 days |
+    # | 31536000 | 365 days |
+    #
+    # Overview of Cache-control values:
+    #
+    #     max-age=<seconds>
+    #         The maximum amount of time a resource is considered fresh. Unlike
+    #         Expires, this directive is relative to the time of the request.
+    #
+    #     s-maxage=<seconds>
+    #         Overrides max-age or the Expires header, but only for shared
+    #         caches (e.g., proxies). Ignored by private caches.
+    #
+    #     More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control
+    #
+    # Our Cache-Control header:
+    #
+    # * It tells all caches (both proxies like Cloudflare and the users web browser) that the asset can be cached.
+    # * It tells shared caches (e.g. Cloudflare) that they can cache it for 365 days
+    # * It tells browsers that they should cache for 365 days
+    #
+    # Cloudflare will respect s-maxage if it is set so change that value if you
+    # want Cloudflare to cache differently than then browser.
+    #
+    config.public_file_server.headers = {
+      "Cache-Control" => "public, s-maxage=31536000, max-age=31536000",
+      "Expires" => "#{1.year.from_now.to_formatted_s(:rfc822)}"
```

Comment:
It might be worth clarifying in the comments above that with this being an iterpolated string, out of the possible options from:

1. 1 year from the date of this release
2. 1 year from the time the asset was first requested by the client
3. 1 year from the time this file was last evaluated

It is going to be the _last_ value that applies here.

Reply:
This seems good. I'm going to do some research about whether we need `Expires` at all.